src/core/folder.h

```diff
@@ -179,6 +183,8 @@ private Q_SLOTS:
     bool defer_content_test : 1;
 
     static std::unordered_map<FilePath, std::weak_ptr<Folder>, FilePathHash> cache_;
+    static FilePath cutFilesDirPath_;
+    static std::shared_ptr<const HashSet> cutFilesHashSet_;
```

Comment:
> I'm here to support (especially) my code.

My solution will be just a suggestion that you could disagree about; I won't touch your code. And I also support it.